scripts/linters/pylint_extensions.py

```diff
@@ -2139,6 +2141,64 @@ def visit_classdef(self, node):
             self.add_message('no-dunder-metaclass', node=node)
 
 
+class StringConcatenationChecker(checkers.BaseChecker):
+    """Custom pylint checker which prohibits use of string concatenation and
+     promotes string interpolation instead.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+
+    name = 'no-string-concatenation'
+    priority = -1
+    msgs = {
+        'C0035': (
+            'Please replace string concatenation with string interpolation.',
+            'string-interpolation',
+            'Enforce use of string interpolation over string concatenation.',
+        ),
+        'C0036': (
+            'Please remove the concatenation as it is not needed',
+            'no-string-concatenation',
+            'Enforce removal of unnecessary string concatenation.',
+        ),
+    }
+
+    def is_node_string_constant(self, node):
+        """Helper function for visit_binop. Called to check if any of the
+        operand is a string constant.
```

Comment:
This looks like a great helper function! :)

Add args and returns docstring*

Reply:
I have added args and returns. Thanks!

Reply:
Done!